command/agent/template/template.go

```diff
@@ -255,16 +267,26 @@ func newRunnerConfig(sc *ServerConfig, templates ctconfig.TemplateConfigs) (*ctc
 		ServerName: pointerutil.StringPtr(""),
 	}
 
-	if strings.HasPrefix(sc.VaultConf.Address, "https") || sc.VaultConf.CACert != "" {
-		skipVerify := sc.VaultConf.TLSSkipVerify
+	if strings.HasPrefix(*conf.Vault.Address, "https") || sc.AgentConfig.Vault.CACert != "" {
+		skipVerify := sc.AgentConfig.Vault.TLSSkipVerify
 		verify := !skipVerify
 		conf.Vault.SSL = &ctconfig.SSLConfig{
 			Enabled: pointerutil.BoolPtr(true),
 			Verify:  &verify,
-			Cert:    &sc.VaultConf.ClientCert,
-			Key:     &sc.VaultConf.ClientKey,
-			CaCert:  &sc.VaultConf.CACert,
-			CaPath:  &sc.VaultConf.CAPath,
+			Cert:    &sc.AgentConfig.Vault.ClientCert,
+			Key:     &sc.AgentConfig.Vault.ClientKey,
+			CaCert:  &sc.AgentConfig.Vault.CACert,
+			CaPath:  &sc.AgentConfig.Vault.CAPath,
+		}
+
+		// Only configure TLS Skip Verify if CT is not going through the cache. We can
+		// skip verification if its using the cache because they're part of the same agent.
+		// Agent listener doesn't support mTLS listeners.
```

Comment:
The listener stanza supports [all standard listener params](https://www.vaultproject.io/docs/configuration/listener/tcp#tls_client_ca_file), including mTLS params.

Reply:
Interesting because the cache documentation doesn't have them listed: https://www.vaultproject.io/docs/agent/caching#configuration-listener.

Reply:
The docs on that page might be incomplete. `StartNewListener` [uses all of the TLS params](https://github.com/hashicorp/vault/blob/f5be0716dbe41ea5b909d3c8a81ebac37fd124e0/command/agent/cache/listener.go#L60-L66) that it accepts and doesn't seem to do any special handling on the agent side.

The listener stanza [section](https://www.vaultproject.io/docs/agent#listener-stanza) on Agent also mentions that it accepts all of the standard listener configuration.

Reply:
Removed the comment, thanks for clarifying.